h/api/test/storage_test.py

```diff
@@ -627,6 +623,74 @@ def db(self):
         return mock.Mock(spec=db.Session)
 
 
+@pytest.mark.usefixtures('models',
+                         'update_document_metadata')
+class TestUpdateAnnotation(object):
+
+    def test_it_calls_get(self, annotation_data, models):
+        storage.update_annotation(mock.Mock(),
+                                  'test_annotation_id',
+                                  annotation_data)
+
+        models.Annotation.query.get.assert_called_once_with(
+            'test_annotation_id')
+
+    def test_it_updates_the_annotation(self, annotation_data, models):
+        annotation = models.Annotation.query.get.return_value = mock.Mock()
+
+        storage.update_annotation(mock.Mock(),
+                                  'test_annotation_id',
+                                  annotation_data)
+
+        for key, value in annotation_data.items():
+            assert getattr(annotation, key) == value
```

Comment:
This and the test above might benefit from going to the database directly and not using mocks. Otherwise looks good.